common/config/config_test.go

```diff
@@ -401,7 +401,72 @@ func TestEnvironmentVariables(t *testing.T) {
 
 	require.NoError(t, os.Setenv("A_VARIABLE_THAT_DOES_NOT_HAVE_PREFIX", "should be ignored"))
 
-	foo, err := ParseConfig(common.TestLogger(t), DefaultFoo(), "PREFIX", nil, configFile)
+	foo, err := ParseConfig(common.TestLogger(t), DefaultFoo(), "PREFIX", nil, nil, configFile)
+	require.NoError(t, err)
+
+	// Verify that environment variables have overridden the config file values.
+	require.Equal(t, "value from env var", foo.String) // from env
+	require.Equal(t, -999, foo.Int)                    // from env
+	require.Equal(t, int64(0), foo.Int64)              // from env (zero value)
+	require.Equal(t, int32(0), foo.Int32)              // from env (zero value)
+	require.Equal(t, int16(4), foo.Int16)              // from config
+	require.Equal(t, int8(5), foo.Int8)                // from config
+	require.Equal(t, uint(6), foo.Uint)                // from config
+	require.Equal(t, uint64(7), foo.Uint64)            // from config
+	require.Equal(t, uint32(8), foo.Uint32)            // from config
+	require.Equal(t, uint16(9), foo.Uint16)            // from config
+	require.Equal(t, uint8(10), foo.Uint8)             // from config
+	require.Equal(t, 11.11, foo.Float64)               // from config
+	require.Equal(t, float32(12.12), foo.Float32)      // from config
+	require.Equal(t, 5*time.Second, foo.Duration)      // from config
+	require.Equal(t, false, foo.Bool)                  // from config
+
+	// Bar field
+	require.Equal(t, "bar A", foo.Bar.A) // from config
+	require.Equal(t, -777, foo.Bar.B)    // from env
+	require.Equal(t, true, foo.Bar.C)    // from config
+
+	// Bar.Baz field
+	require.NotNil(t, foo.Bar.Baz)
+	require.Equal(t, "env var bar baz X", foo.Bar.Baz.X) // from env
+	require.Equal(t, 444, foo.Bar.Baz.Y)                 // from env
+	require.Equal(t, false, foo.Bar.Baz.Z)               // from env
+
+	// Baz field - the env vars use FOO_BAZ_PARTIAL_* which doesn't match foo.Baz,
+	// so these should come from config
+	require.NotNil(t, foo.Baz)
+	require.Equal(t, "baz X", foo.Baz.X) // from config
+	require.Equal(t, 27, foo.Baz.Y)      // from config
+	require.Equal(t, true, foo.Baz.Z)    // from config
+}
+
+func TestAliasedEnvironmentVariables(t *testing.T) {
```

Comment:
Do you have a test anywhere for when both the old and new vars are set? I don't see one